mobile_verifier/src/banning/db.rs

```diff
@@ -8,20 +8,31 @@ use sqlx::{PgConnection, PgPool, Row};
 
 use super::BAN_CLEANUP_DAYS;
 
+// When retreiving banned radios, we want to get the
+// latest ban for a radio at the given time.
+//
+// If a radio was banned for POC yesterday,
+// and today a ban for DATA comes in.
+// Running files from yesterday, the radio should not be banned.
 pub async fn get_banned_radios(
     pool: &PgPool,
     timestamp: DateTime<Utc>,
 ) -> anyhow::Result<HashSet<PublicKeyBinary>> {
     let banned = sqlx::query(
         r#"
-            SELECT hotspot_pubkey
+        WITH latest AS (
+            SELECT DISTINCT ON (hotspot_pubkey) *
             FROM hotspot_bans
-            WHERE 
-                ban_type in ('all', 'poc')
-                AND received_timestamp <= $1
-                AND (expiration_timestamp IS NULL 
-                    OR expiration_timestamp >= $1)
-            "#,
+            WHERE
+                received_timestamp <= $1
```

Comment:
should it be
`received_timestamp < $1`
and
`expiration_timestamp >= $1`

Reply:
Could be. The way I read it, you are banned from the moment we receive a ban `<=`, up to when it expires `>` and not a nanosecond more. 👯

Reply:
Fixed with [f4c6a0f](https://github.com/helium/oracles/pull/984/commits/f4c6a0fb284decf1b865112caf2bb056a05cb67f)